### PR TITLE
Add project overview.

### LINKED ISSUE #257: Add a project overview
To align the project, an explicit project overview should be maintained.  Minimally it should contain:

- The problem being addressed
- The project's goal
- The project's scope

Additionally, some mention of the development methodology and project status is a good idea.

### PR DESCRIPTION
**Pull request by edwbuck**
Adds a project overview allowing contributors to align more easily.

closes #257

### CODE REVIEW

**Review by kfox1111 — APPROVED** [review]
This looks good to me. Nothing objectionable. It is general enough that we can deal with details as we need to later.

**Comment by kfox1111** [comment]
@marcofranssen @faisal-memon What do you think?

**Review by anvega — APPROVED** [review]

**Review by mrsabath — APPROVED** [review]
This is a good starting point, LGTM!

File: project/overview.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<!-- vim: ft=markdown colorcolumn=72
+-->
+# Project Overview
+
+The spire/helm-chart effort intends to deliver a set of helm charts to
+support SPIRE deployments in Kubernetes to widest possible audience of
```
Comment (faisal-memon) [inline project/overview.md:6]:
```suggestion
support SPIRE deployments in Kubernetes to the widest possible audience of
```

File: project/overview.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+users.
+
+## Mission and Scope
+This mission of this effort is to provide a stable upstream repository
```
Comment (faisal-memon) [inline project/overview.md:10]:
```suggestion
The mission of this effort is to provide a stable upstream repository
```

File: project/overview.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+of Helm Charts sufficient for any organization or individual to use.
+
+### What problem does this project address?
+Prior to this effort, no single authoritative repository for SPIRE Helm
+Charts existed, imposing an additional burden on SPIRE adoption as each
+organization or individual had to develop their own Kubernetes
+deployment packaging.
+
+### What is the goal of this project?
+The goal of this project is to be the primary means of deployment for
```
Comment (faisal-memon) [inline project/overview.md:20]:
```suggestion
The goal of this project is to be a means of deployment for
```

File: project/overview.md (hunk 2 above)
Comment (edwbuck) [reply project/overview.md:10]: Not a fan of referring to "this" before actually stating what this is in a concrete way.

File: project/overview.md (hunk 3 above)
Comment (edwbuck) [reply project/overview.md:20]: We have the branding, we have the lineage, and goals are aspirational, so let's shoot for the stars.  If we miss, at least our arrow will have travelled further.

**Review by faisal-memon — APPROVED** [review]

File: project/overview.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+-->
+# Project Overview
+
+The spire/helm-chart effort intends to deliver a set of helm charts to
+support SPIRE deployments in Kubernetes to the widest possible audience of
+users.
+
+## Mission and Scope
+The mission of this effort is to provide a stable upstream repository
+of Helm Charts sufficient for any organization or individual to use.
```
Comment (marcofranssen) [inline project/overview.md:11]: Satisfy markdown linter

```suggestion

The mission of this effort is to provide a stable upstream repository
of Helm Charts sufficient for any organization or individual to use.
```

File: project/overview.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+### What problem does this project address?
+Prior to this effort, no single authoritative repository for SPIRE Helm
```
Comment (marcofranssen) [inline project/overview.md:14]:
```suggestion

Prior to this effort, no single authoritative repository for SPIRE Helm
```

File: project/overview.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+Charts existed, imposing an additional burden on SPIRE adoption as each
+organization or individual had to develop their own Kubernetes
+deployment packaging.
+
+### What is the goal of this project?
+The goal of this project is to be the primary means of deployment for
```
Comment (marcofranssen) [inline project/overview.md:20]:
```suggestion

The goal of this project is to be the primary means of deployment for
```

File: project/overview.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+SPIRE into Kubernetes clusters, providing a flexible solution that can
+be used in a wide number of different scenarios.
+
+### What is the scope of this project?
+The scope includes user documentation, chart creation, chart delivery,
```
Comment (marcofranssen) [inline project/overview.md:25]:
```suggestion

The scope includes user documentation, chart creation, chart delivery,
```

File: project/overview.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+chart testing, and chart maintenance of a set of Helm Charts sufficient
+for deploying SPIRE in any SPIRE documented configuration, including
+direction on how to extend the Charts to integrate non-SPIRE plugins.
+
+### What development methodology is used?
+An Open Source Development methodology is used, where issues are tracked
```
Comment (marcofranssen) [inline project/overview.md:31]:
```suggestion

An Open Source Development methodology is used, where issues are tracked
```

File: project/overview.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+within the primary github repository.
+
+## Status
+The project is currently in a pre-release status.  While the standalone
```
Comment (marcofranssen) [inline project/overview.md:35]:
```suggestion

The project is currently in a pre-release status.  While the standalone
```

File: project/overview.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+deployment is close to release quality, the tiered deployment is not.
```
Comment (marcofranssen) [inline project/overview.md:36]: We might want to more explicitly define what `tiered` deployment means (nested and federated).

**Pull request merged**